python/kserve/kserve/protocol/rest/server.py

```diff
@@ -103,55 +111,36 @@ def create_application(self):
         self.app.add_exception_handler(ServerNotReady, server_not_ready_handler)
         self.app.add_exception_handler(Exception, generic_exception_handler)
 
-
-class UvicornServer:
-    def __init__(
-        self,
-        app: FastAPI,
-        http_port: int,
-        data_plane: DataPlane,
-        model_repository_extension,
-        log_config: Optional[Union[str, Dict]] = None,
-        access_log_format: Optional[str] = None,
-        workers: int = 1,
-    ):
-        super().__init__()
-        rest_server = RESTServer(app, data_plane, model_repository_extension)
-        rest_server.create_application()
-        app.add_middleware(
+    def _add_middlewares(self):
+        self.app.add_middleware(
             TimingMiddleware,
             client=PrintTimings(),
-            metric_namer=StarletteScopeToName(prefix="kserve.io", starlette_app=app),
-        )
-        self.cfg = uvicorn.Config(
-            app="kserve.model_server:app",
-            host="0.0.0.0",
-            log_config=log_config,
-            port=http_port,
-            workers=workers,
+            metric_namer=StarletteScopeToName(
+                prefix="kserve.io", starlette_app=self.app
+            ),
         )
 
         # More context in https://github.com/encode/uvicorn/pull/947
         # At the time of writing the ASGI specs are not clear when it comes
         # to change the access log format, and hence the Uvicorn upstream devs
         # chose to create a custom middleware for this.
         # The allowed log format is specified in https://github.com/Kludex/asgi-logger#usage
-        if access_log_format:
+        if self.access_log_format:
             from asgi_logger import AccessLoggerMiddleware
 
             # As indicated by the asgi-logger docs, we need to clear/unset
             # any setting for uvicorn.access to avoid log duplicates.
             logging.getLogger("uvicorn.access").handlers = []
-            app.add_middleware(AccessLoggerMiddleware, format=access_log_format)
+            self.app.add_middleware(
+                AccessLoggerMiddleware, format=self.access_log_format
+            )
             # The asgi-logger settings don't set propagate to False,
             # so we get duplicates if we don't set it explicitly.
             logging.getLogger("access").propagate = False
 
-        self.server = _NoSignalUvicornServer(config=self.cfg)
-
-    async def run(self):
-        await self.server.serve()
-
-    async def stop(self, sig: Optional[int] = None):
```

Comment:
Uvicron registers its [own signal handler and re raises the signal](https://github.com/encode/uvicorn/blob/bfa754e21e2cc1d5b0d7cabf24933a6c3afc315e/uvicorn/server.py#L315). So we don't have to handle the shutdown process.